tests/integration/zome_call.rs

```diff
@@ -0,0 +1,87 @@
+use base64::prelude::*;
+use fixt::prelude::*;
+use holochain::fixt::DnaHashFixturator;
+use holochain_http_gateway::{
+    config::{AllowedFns, Configuration},
+    tracing::initialize_tracing_subscriber,
+};
+use reqwest::StatusCode;
+
+use crate::TestApp;
+
+#[tokio::test]
+async fn zome_call_uses_correct_route_parameters() {
+    initialize_tracing_subscriber("info");
+
+    let app = TestApp::spawn().await;
+
+    let dna_hash = fixt!(DnaHash);
+    let coordinator = "coord98765";
+    let zome = "custom_zome";
+    let function = "special_function";
+
+    let response = app
+        .client
+        .get(format!(
+            "http://{}/{}/{}/{}/{}",
+            app.address, dna_hash, coordinator, zome, function
+        ))
+        .send()
+        .await
+        .expect("Failed to execute request");
+
+    assert_eq!(response.status(), StatusCode::OK);
+}
+
+#[tokio::test]
+async fn zome_call_with_payload_exceeding_limit_fails() {
+    initialize_tracing_subscriber("info");
+
+    let mut allowed_fns = std::collections::HashMap::new();
+    allowed_fns.insert("forum".to_string(), AllowedFns::All);
+
+    // Custom configuration with a very small payload limit
+    let config =
+        Configuration::try_new("ws://localhost:50350", "10", "forum", allowed_fns).unwrap();
+
+    let app = TestApp::spawn_with_config(config).await;
+
+    let large_payload = r#"{"limit":100,"offset":0,"filters":{"author":"user123","tags":["important","featured","latest"],"content_contains":"search term","date_range":{"from":"2023-01-01","to":"2023-12-31"}}"#;
+    let encoded_payload = BASE64_STANDARD.encode(large_payload);
+
+    let response = app
+        .client
+        .get(format!(
+            "http://{}/12345/coordinator/forum/get_posts?payload={}",
+            app.address, encoded_payload
+        ))
+        .send()
+        .await
+        .expect("Failed to execute request");
+
+    assert_eq!(response.status(), StatusCode::BAD_REQUEST);
+}
+
+#[tokio::test]
+async fn zome_call_with_small_payload_works() {
```

Comment:
This kind of test where we're not needing a full integration setup might be a good place to look at Axum's other testing options. We just want to run the middleware and parse a request. We don't need networking and a real client/server setup.

Especially as we need some more tests. At least one that checks for an invalid DNA hash. Then ideally one that sends a payload that isn't valid JSON, though that depends on a discussion above.

Reply:
Add some extra tests for checking for valid json from the decoded payload, and a test to verify that we can derive the `DnaHash` from the URL params

Reply:
I wasn't clear enough there. I linked to a specific test as an example of making the test non-flaky without a sleep. That same file contains a bunch of other tests though. If you've looked at those and decided that there's no value there then that's fine, I just want to make sure I've clearly communicated that there are some other examples of just testing a router without booting up a whole server/client.

Reply:
What is the outcome here, are we sticking with the current testing pattern?

Reply:
Yea, I am fine to stick with it, especially after the refactor to how the TCP listern is handled